test/unit/app/models/edition/custom_lead_image_test.rb

```diff
@@ -3,11 +3,18 @@
 class Edition::CustomLeadImageTest < ActiveSupport::TestCase
   include ActionDispatch::TestProcess
 
+  def stubbed_image(name)
+    build(:image).tap do |image|
+      image.stubs(:filename).returns(name)
+      image.stubs(:url).returns("http://example.com/#{name}")
```

Comment:
Do we need to stub this out? Can we not just create an image with the filename we want?

Reply:
I tried avoiding a stub in the first place, but it looks like the [Image](https://github.com/alphagov/whitehall/blob/ff071c7f2dfe5b9e949216d6a9fee08bfdc894f4/test/factories/images.rb) and [ImageData](https://github.com/alphagov/whitehall/blob/ff071c7f2dfe5b9e949216d6a9fee08bfdc894f4/test/factories/image_data.rb) factories have been changed with the recent work done by the Assets team.

From what I can tell, the factory can only create an SVG image or a JPEG image. And the filename isn't configurable – it seems like the JPEG is hardcoded as `minister-of-funk.960x640.jpg` and the SVG is `test-svg.svg`.

Really I want to create two 'normal' images (e.g. not SVGs, because we know they sometimes get treated differently) that have different filenames.

So I figured this stub was an easy way to avoid changing how that factory works. (Perhaps that's a bit lazy of me!?)

Reply:
Yea i've seen quite a lot of stuff that's not used the factories which probably isn't ideal (if you search for `build(:image_data)` you'll see what i mean. This seems fine to me though.